compiler/rustc_parse/src/lexer/mod.rs

```diff
@@ -140,8 +143,8 @@ impl<'a> StringReader<'a> {
     /// Turns simple `rustc_lexer::TokenKind` enum into a rich
     /// `librustc_ast::TokenKind`. This turns strings into interned
     /// symbols and runs additional validation.
-    fn cook_lexer_token(&self, token: rustc_lexer::TokenKind, start: BytePos) -> TokenKind {
-        match token {
+    fn cook_lexer_token(&self, token: rustc_lexer::TokenKind, start: BytePos) -> Option<TokenKind> {
+        let res = match token {
```

Comment:
```suggestion
        Some(match token {
```

Reply:
Urgh, thanks, sorry for not remembering that this is the preferred style in rustc!